Instagram.js

```diff
@@ -22,50 +22,46 @@ const patchPostMessageJsCode = `(${String(function () {
   window.postMessage = patchedPostMessage;
 })})();`;
 
-export default class Instagram extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      modalVisible: false,
-      key: 1,
-    };
-  }
+export default function Instagram(props) {
+  const { wrapperStyle, containerStyle, closeStyle, onLoginSuccess, onLoginFailure, redirectUrl, appId, appSecret, responseType, scopes, language = 'en', incognito = false } = props;
+  const [modalVisible, setModalVisible] = useState(false);
+  const [key, setKey] = useState(1);
+  const webViewRef = useRef(null);
 
-  show() {
-    this.setState({ modalVisible: true });
+  function show() {
+    setModalVisible(true);
   }
 
-  hide() {
-    this.setState({ modalVisible: false });
+  function hide() {
+    setModalVisible(false);
   }
 
-  async onNavigationStateChange(webViewState) {
+  async function onNavigationStateChange(webViewState) {
     const { url } = webViewState;
-    const { key } = this.state;
+
     if (
       webViewState.title === 'Instagram' &&
       webViewState.url === 'https://www.instagram.com/'
     ) {
-      this.setState({ key: key + 1 });
+      setKey((k) => k + 1);
```

Comment:
why exactly do you need to increase the key counter on that? just to force a rerender?

Reply:
I didn't understand too

Should we do this change in this PR, since its only goal is to move from class to FC?